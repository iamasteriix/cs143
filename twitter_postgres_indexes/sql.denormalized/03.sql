SELECT
    lang,
    count(*)
FROM (
    SELECT
        DISTINCT t1.data ->> 'id' AS id_tweets,
        t1.data ->> 'lang' AS lang
    FROM tweets_jsonb t1
    WHERE
        t1.data -> 'entities' -> 'hashtags' @@ '$[*].text == "coronavirus"'
        OR t1.data -> 'extended_tweet' -> 'entities' -> 'hashtags' @@ '$[*].text == "coronavirus"'
    ) t
GROUP BY (1)
ORDER BY count DESC, lang;
