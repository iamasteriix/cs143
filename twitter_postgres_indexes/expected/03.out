 lang | count 
------+-------
 en   |  1616
 es   |   664
 und  |   330
 pt   |   325
 de   |   155
 tr   |   146
 it   |   139
 fr   |   114
 in   |    88
 nl   |    79
 hi   |    57
 ar   |    22
 zh   |    12
 ca   |    11
 ja   |     9
 tl   |     9
 el   |     8
 bn   |     6
 da   |     6
 uk   |     6
 mr   |     4
 pl   |     4
 et   |     3
 ta   |     3
 th   |     3
 fa   |     2
 fi   |     2
 gu   |     2
 lt   |     2
 si   |     2
 sv   |     2
 cy   |     1
 eu   |     1
 ht   |     1
 hu   |     1
 is   |     1
 iw   |     1
 ko   |     1
 no   |     1
 ru   |     1
 sr   |     1
(41 rows)

